src/lib/utility/calendar.js

```diff
@@ -247,110 +266,105 @@ export function collision(a, b, lineHeight, collisionPadding = EPSILON) {
+  // calculate non-overlapping positions
+  let curHeight = groupHeight
+  let verticalMargin = lineHeight - item.dimensions.height
+  if (item.dimensions.stack && item.dimensions.top === null) {
+    item.dimensions.top = totalHeight + verticalMargin
+    curHeight = Math.max(curHeight, lineHeight)
+    do {
+      var collidingItem = null
+      //Items are placed from i=0 onwards, only check items with index < i
+      for (var j = i-1, jj = 0; j >= jj; j--) {
+        var other = group[j]
+        if (
+          other.dimensions.top !== null &&
+          other.dimensions.stack &&
+          collision(item.dimensions, other.dimensions, lineHeight)
+        ) {
+          collidingItem = other
+          break
+        } else {
+          // console.log('dont test', other.top !== null, other !== item, other.stack);
+        }
       }
-    }
 
-    if (groupVal.height) {
-      groupHeights.push(groupVal.height)
-      totalHeight += groupVal.height
-    } else {
-      groupHeights.push(Math.max(groupHeight + verticalMargin, lineHeight))
-      totalHeight += Math.max(groupHeight + verticalMargin, lineHeight)
-    }
-  })
-  return {
-    height: totalHeight,
-    groupHeights,
-    groupTops
+      if (collidingItem != null) {
+        // There is a collision. Reposition the items above the colliding element
+        item.dimensions.top = collidingItem.dimensions.top + lineHeight
+        curHeight = Math.max(
+          curHeight,
+          item.dimensions.top + item.dimensions.height - totalHeight
+        )
+      }
+    } while (collidingItem)
   }
+  return {groupHeight: curHeight, verticalMargin}
 }
 
-export function nostack(items, groupOrders, lineHeight, groups) {
-  var i, j=0, iMax
+// Calculate the position of this item for a group that is not being stacked
+function groupNoStack(lineHeight, item, groupHeight, totalHeight) {
+  let verticalMargin = (lineHeight - item.dimensions.height) / 2
+  if (item.dimensions.top === null) {
+    item.dimensions.top = totalHeight + verticalMargin
+    groupHeight = Math.max(groupHeight, lineHeight)
+  }
+  return {groupHeight, verticalMargin: 0}
+}
 
+/**
+ * Stack all groups
+ * @param {*} items items to be stacked
+ * @param {*} groupOrders the groupOrders object
+ * @param {*} lineHeight 
+ * @param {*} stackItems should items be stacked?
+ */
+export function stackAll(items, groupOrders, lineHeight, stackItems) {
+  var i, iMax
   var totalHeight = 0
 
   var groupHeights = []
   var groupTops = []
 
   var groupedItems = getGroupedItems(items, groupOrders)
 
-  groupedItems.forEach(function(group) {
-    var groupVal = groups[j++];
-
-    // calculate new, non-overlapping positions
+  for (var index in groupedItems) {
+    const groupItems = groupedItems[index]
+    const {items, group} = groupItems
     groupTops.push(totalHeight)
-
+  
+    // Is group being stacked?
+    const isGroupStacked = group.stackItems !== undefined ? group.stackItems : stackItems
```

Comment:
@Ilaiwi This is where the logic for the group stacking is decided. The groups array is truthy.